internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	}
+
+	scopeAttrTypes := map[string]attr.Type{
+		"name":        basetypes.StringType{},
+		"description": basetypes.StringType{},
+		"dynamic":     basetypes.BoolType{},
+	}
+	state.Scopes, _ = types.SetValueFrom(ctx, types.ObjectType{AttrTypes: scopeAttrTypes}, toStateScopes)
+
+	// state.ScopeGroups
+	scopeGroups := r.GetScopeGroups()
+	toStateScopeGroups := []client.ScopeGroupEntry{}
+	for _, scopeGroup := range scopeGroups {
+		scopeGroupEntry := client.ScopeGroupEntry{}
+		scopeGroupEntry.Name = scopeGroup.Name
+		scopeGroupEntry.Description = scopeGroup.Description
+		scopeGroupEntry.Scopes = scopeGroup.Scopes
+		toStateScopeGroups = append(toStateScopeGroups, scopeGroupEntry)
+	}
+
+	scopeGroupAttrTypes := map[string]attr.Type{
+		"name":        basetypes.StringType{},
+		"description": basetypes.StringType{},
+		"scopes":      basetypes.SetType{ElemType: types.StringType},
+	}
+	state.ScopeGroups, _ = types.SetValueFrom(ctx, types.ObjectType{AttrTypes: scopeGroupAttrTypes}, toStateScopeGroups)
+
+	// state.ExclusiveScopes
+	exclusiveScopes := r.GetExclusiveScopes()
+	toStateExclusiveScopes := []client.ScopeEntry{}
+	for _, exclusiveScope := range exclusiveScopes {
+		exclusiveScopeEntry := client.ScopeEntry{}
+		exclusiveScopeEntry.Name = exclusiveScope.Name
+		exclusiveScopeEntry.Description = exclusiveScope.Description
+		exclusiveScopeEntry.Dynamic = exclusiveScope.Dynamic
+		toStateExclusiveScopes = append(toStateExclusiveScopes, exclusiveScopeEntry)
+	}
+
+	exclusiveScopeAttrTypes := map[string]attr.Type{
+		"name":        basetypes.StringType{},
+		"description": basetypes.StringType{},
+		"dynamic":     basetypes.BoolType{},
+	}
+	state.ExclusiveScopes, _ = types.SetValueFrom(ctx, types.ObjectType{AttrTypes: exclusiveScopeAttrTypes}, toStateExclusiveScopes)
+
+	// state.ScopeGroups
+	exclusiveScopeGroups := r.GetExclusiveScopeGroups()
+	toStateExclusiveScopeGroups := []client.ScopeGroupEntry{}
+	for _, exclusiveScopeGroup := range exclusiveScopeGroups {
+		exclusiveScopeGroupEntry := client.ScopeGroupEntry{}
+		exclusiveScopeGroupEntry.Name = exclusiveScopeGroup.Name
+		exclusiveScopeGroupEntry.Description = exclusiveScopeGroup.Description
+		exclusiveScopeGroupEntry.Scopes = exclusiveScopeGroup.Scopes
+		toStateExclusiveScopeGroups = append(toStateExclusiveScopeGroups, exclusiveScopeGroupEntry)
+	}
+
+	exclusiveScopeGroupAttrTypes := map[string]attr.Type{
+		"name":        basetypes.StringType{},
+		"description": basetypes.StringType{},
+		"scopes":      basetypes.SetType{ElemType: types.StringType},
+	}
+	state.ExclusiveScopeGroups, _ = types.SetValueFrom(ctx, types.ObjectType{AttrTypes: exclusiveScopeGroupAttrTypes}, toStateExclusiveScopeGroups)
+
+	// state.PersistentGrantContract
+	getPersistentGrantContract := r.GetPersistentGrantContract()
+	// Build core attributes
+	nameAttributeType := map[string]attr.Type{
+		"name": basetypes.StringType{},
+	}
+	coreAttrs := getPersistentGrantContract.CoreAttributes
```

Comment:
Is there any chance this could be nil?

Reply:
these are the values that are always returned from the client